src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -92,20 +94,69 @@ def check_extracted_metadata(self, metadata: dict):
+        """
+        Just to ensure each interface can call .get_original_timestamps() without an error raising.
+
+        Also, that it always returns non-empty.
+        """
+        timestamps = self.interface.get_original_timestamps()
+
+        assert len(timestamps) != 0
+
+    def check_get_timestamps(self):
+        """
+        Just to ensure each interface can call .get_timestamps() without an error raising.
+
+        Also, that it always returns non-empty.
+        """
+        timestamps = self.interface.get_timestamps()
+
+        assert len(timestamps) != 0
+
+    def check_align_starting_time_internal(self):
+        fresh_interface = self.data_interface_cls(**self.test_kwargs)
+        unaligned_timestamps = fresh_interface.get_timestamps()
+
+        starting_time = 1.23
+        fresh_interface.align_starting_time(starting_time=starting_time)
+
+        aligned_timestamps = fresh_interface.get_timestamps()
+        expected_timestamps = unaligned_timestamps + starting_time
+        assert_array_equal(x=aligned_timestamps, y=expected_timestamps)
+
+    def check_align_timestamps_internal(self):
+        unaligned_timestamps = self.interface.get_original_timestamps()
+
+        aligned_timestamps = unaligned_timestamps + 1.23 + np.random.random(size=unaligned_timestamps.shape)
+        self.interface.align_timestamps(aligned_timestamps=aligned_timestamps)
+
+        retrieved_aligned_timestamps = self.interface.get_timestamps()
+        assert_array_equal(x=retrieved_aligned_timestamps, y=aligned_timestamps)
+
+    def check_align_starting_time_external(self):
+        pass  # TODO: generalize
+
     def test_conversion_as_lone_interface(self):
         interface_kwargs = self.interface_kwargs
         if isinstance(interface_kwargs, dict):
             interface_kwargs = [interface_kwargs]
         for num, kwargs in enumerate(interface_kwargs):
             with self.subTest(str(num)):
                 self.case = num
-                self.interface = self.data_interface_cls(**kwargs)
+                self.test_kwargs = kwargs
+                self.interface = self.data_interface_cls(**self.test_kwargs)
                 self.check_metadata_schema_valid()
                 self.check_conversion_options_schema_valid()
                 self.check_metadata()
-                nwbfile_path = str(self.save_directory / f"{self.data_interface_cls.__name__}_{num}.nwb")
-                self.run_conversion(nwbfile_path)
-                self.check_read_nwb(nwbfile_path=nwbfile_path)
+                self.nwbfile_path = str(self.save_directory / f"{self.data_interface_cls.__name__}_{num}.nwb")
+                self.run_conversion(nwbfile_path=self.nwbfile_path)
+                self.check_read_nwb(nwbfile_path=self.nwbfile_path)
+
+                # Temporal alignment checks
+                self.check_get_timestamps()
+                self.check_align_starting_time_internal()
+                self.check_align_starting_time_external()
```

Comment:
This adds some essential checks for all interfaces using the new mixin

Reply:
Everything except a handful of behavior interfaces not currently using this mixin, yes. See below for the temporary child-class overrides that disable it in failing cases